core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2222,6 +2223,9 @@ private[spark] object Utils extends Logging {
         isBindCollision(e.getCause)
       case e: MultiException =>
         e.getThrowables.asScala.exists(isBindCollision)
+      case e: NativeIoException =>
+        e.getMessage().startsWith("bind() failed: ") ||
```

Comment:
From the netty code, the return of `NativeIoException.getMessage` is not null.
The only way to  create a NativeIoException instance is to call [Errors.newIOException](https://github.com/netty/netty/blob/4.0/transport-native-epoll/src/main/java/io/netty/channel/unix/Errors.java#L116)